src/BaseStyles.tsx

```diff
@@ -26,19 +26,24 @@ const GlobalStyle = createGlobalStyle`
   }
 
 `
-const Base = props => {
+
+const Base = styled.div<SystemTypographyProps & SystemCommonProps>`
+  ${TYPOGRAPHY};
+  ${COMMON};
+`
+
+export type BaseStylesProps = React.ComponentProps<typeof Base>
```

Comment:
Hmm, `BaseStylesProps` doesn't seem to include `SystemTypographyProps & SystemCommonProps`. I wonder if `React.ComponentProps` doesn't work with styled-components. If so, all the components we've migrated so far might be exporting incorrect props types  😬

Reply:
Writing a utility type to replace `React.ComponentProps` solved this issue. The utility type was copied from https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase#wrappingmirroring-a-component